codex/utils/fileutils.nim

```diff
@@ -49,6 +49,12 @@ proc secureWriteFile*[T: byte|char](path: string,
   else:
     writeFile(path, data, 0o600)
 
+proc checkSecureFile*(path: string): IOResult[bool] =
```

Comment:
Hmm, looking at how `checkAndCreateDataDir` is implemented in `fileutils`, it does a lot more than the ACL check. I'm not 100% sure if that is needed, but it's worth understanding why.

Reply:
Perhaps I'm missing something, but I don't see anything in there that we need. It's complexity seems to be mostly because it deals with a directory instead of a file, and that it tries to create it when it doesn't exist.